pkg/controller/migmigration/migrate.go

```diff
@@ -93,42 +80,52 @@ func (r *ReconcileMigMigration) migrate(migration *migapi.MigMigration) (int, er
 		return 0, err
 	}
 
-	migration.Status.SetCondition(migapi.Condition{
-		Type:     Running,
-		Status:   True,
-		Reason:   task.Phase.Name,
-		Category: Advisory,
-		Message:  RunningMessage,
-	})
-
 	// Result
-	migration.Status.Phase = task.Phase.Name
-	if task.Phase.Equals(WaitOnResticRestart) {
-		return 10, nil
-	}
-	if task.Phase.Final() {
+	migration.Status.Phase = task.Phase
+
+	// Succeeded
+	if task.Phase == Completed {
 		migration.Status.DeleteCondition(Running)
-		migration.Status.SetCondition(migapi.Condition{
-			Type:     Succeeded,
-			Status:   True,
-			Reason:   task.Phase.Name,
-			Category: Advisory,
-			Message:  SucceededMessage,
-		})
+		failed := task.Owner.Status.FindCondition(Failed)
+		if failed == nil {
+			migration.Status.SetCondition(migapi.Condition{
+				Type:     Succeeded,
+				Status:   True,
+				Reason:   task.Phase,
+				Category: Advisory,
+				Message:  SucceededMessage,
+				Durable:  true,
+			})
+		} else {
+			failed.Status = "True"
+		}
+		return task.Requeue, nil
 	}
-	if task.Phase.Failed() {
+	// Failed
+	if _, found := ErrorPhase[task.Phase]; found {
 		migration.AddErrors(task.Errors)
 		migration.Status.DeleteCondition(Running)
 		migration.Status.SetCondition(migapi.Condition{
 			Type:     Failed,
-			Status:   True,
-			Reason:   task.Phase.Name,
+			Status:   False,
```

Comment:
and here

Reply:
A _durable_ condition is created to capture the failed phase in the _Reason_.  The Status is _False_ until the phase reaches _Complete_.  Then, it is set to _True_.  Since the phases continue beyond the failure phases to affect clean-up, the condition must be created during the failed phase but then the _Status_ is set to _True_ only after the migration has reached _Complete_.